site/docs/snapshot-tag-branch.md

```diff
@@ -0,0 +1,168 @@
+
+| Property                     | Type      | Description |
+|------------------------------|-----------|-------------|
+| **`min-snapshots-to-keep`**  | `int`     | For `branch` type only, the minimum number of snapshots to keep in a branch, default to the current value of table property `history.expire.min-snapshots-to-keep` when this value is evaluated |
+| **`max-snapshot-age-ms`**    | `long`    | The duration before a snapshot tagged or in a branch could be expired by any automatic snapshot expiration process, default to the current value of table property `history.expire.max-snapshot-age-ms` when this value is evaluated |
+
+#### Policy evaluation mechanism
+
+When a snapshot expiration process starts, it follows the steps described below:
+
+1. form an expiration candidate pool containing all snapshots
+2. for each snapshot reference, evaluate the associated policy and move snapshots out of the candidate pool
+3. apply global retention policy to and move snapshots out of the candidate pool
+4. when multiple snapshots can be chosen to be moved out, newer snapshots win
+4. after evaluation, expire all snapshots that are still in the candidate pool
+
+#### Policy evaluation example
+
+Here is an example for how an Iceberg snapshot expiration procedure evaluates what snapshots to expire.
+
+Suppose we have the following snapshot graph and retention policies configured:
+
+```
+A -> B -> C (main)
+ \      (dev)
+  D -> E (b1)
+   \
+    F -> G (b2)
+     \
+      H (b3)
+```
+
+| Policy Type      | Max Age       | Min to Keep | Snapshots Affected       |
+|------------------|---------------|-------------|--------------------------|
+| global           | 5 hours       | 4           | A, B, C, D, E, F, G, H   |
+| branch/main      | 3 days        | 1           | A, B, C                  |
+| branch/b1        | 2 days        | 2           | A, D, E                  |
+| branch/b2        | 1 day         | 0           | A, D, F, G               |
+| tag/dev          | forever       | N/A         | C                        |
+
+Assume that we have a process continuously running the snapshot expiration procedure, we would have the results below as time progresses:
+
+##### Day 1: F, G, H are expired
+
+On day 1, the global and branch b2 max age has passed, affecting A, D, F, G, H.  
+There is no policy configured for branch b3, so H is expired.
+On branch b2, A and D cannot be expired due to branch b1 policy, so only F and G are expired.
+
+```
+A -> B -> C (main)
+ \      (dev)
+  D -> E (b1)
+```
+
+##### Day 2: no snapshot is expired
+
+On day 2, branch b1 max age has also passed, affecting A, D, E.
+Because branch b1 must keep 2 snapshots, only the oldest snapshot of the branch A could be expired. 
+However, A is also in the main branch, so it cannot be expired either.
+As a result, no snapshot is expired.
+
+``` 
+A -> B -> C (main)
+ \      (dev)
+  D -> E (b1)
```

Comment:
I think it would help if D expired here and the branch had `min-snapshots-to-keep` set to 1.